clarity/src/vm/database/clarity_db.rs

```diff
@@ -964,26 +865,32 @@ impl<'a> ClarityDatabase<'a> {
     /// `get_current_block_height`).
     pub fn get_current_burnchain_block_height(&mut self) -> Result<u32> {
         let cur_stacks_height = self.store.get_current_block_height();
-        let last_mined_bhh = if cur_stacks_height == 0 {
-            return Ok(self.burn_state_db.get_burn_start_height());
-        } else {
-            self.get_index_block_header_hash(cur_stacks_height.checked_sub(1).ok_or_else(
-                || {
-                    InterpreterError::Expect(
-                        "BUG: cannot eval burn-block-height in boot code".into(),
-                    )
-                },
-            )?)?
-        };
 
-        self.get_burnchain_block_height(&last_mined_bhh)
-            .ok_or_else(|| {
-                InterpreterError::Expect(format!(
-                    "Block header hash '{}' must return for provided stacks block height {}",
-                    &last_mined_bhh, cur_stacks_height
-                ))
-                .into()
-            })
+        // In epoch 2, we can only access the burn block associated with the last block
+        if self.get_clarity_epoch_version()? < StacksEpochId::Epoch30 {
```

Comment:
Good idea. I now see the other similar methods there. I see most of these methods use a `match` to make it explicit for each epoch. Is that preferred over just using a `<` so that it needs to be considered for each new epoch?

Reply:
Added method in e29ab98bad462fcfc2a11d0c04d827eadad96258.